### DISCUSSION TITLE
Option to not return default album images.

### DISCUSSION THREAD

**Discussion by Tolriq** [navidrome/navidrome#2236] [Ideas]
As discussed previously and since it seems the feature requests should be done here now.

I'd like to have an option for the subsonic API to not return the default Navidrome images when they are missing for the album cover arts.

**Comment by deluan** [comment]
Which of the 3 images are you talking about:
- getArtistInfo2 urls
- getArtist artistImageUrl
- getArtist coverArt

I think coverArt can't be changed as it may break other clients, have to test. For the other two, the current artwork resolution happens at query time, so Navidrome does not know if the artist have an image or not. What do you suggest? Maybe this could be an implemented as a OpenSubsonic "enhancement"?

**Reply by Tolriq** [reply]
I'm talking about the album images as per title (and songs). For the artists you already fixed this and return 404 after our discussion, You left the album part for later and I think you forget about the request :p

For the albums we talked previously and you said you where open to add a option for that. Return no default coverArt if there's no art.
Since it's an option it's opt in for the users.

**Reply by deluan** [reply]
Ops, sorry, fingers faster than the brain 😛 

> For the albums we talked previously and you said you where open to add a option for that. Return no default coverArt if there's no art.

You mean, return the `coverArt` field empty in the `album` object, or return a "not-found" in the `getCoverArt` endpoint.
The first one is trick, as the current artwork resolution happens at query time, so Navidrome does not know if the album have an image or not until it has to look up for it. 

In `getCoverArt` I could implement something like this, probably a config in the individual player. 

Again, I think you should open this for discussion in the OpenSubsonic forum.

**Reply by Tolriq** [reply]
Either one would work. For the record you are one of the few to always return your default images.

And yes don't worry I'll open a lot of things to define things after the vote and the tooling is in place. Just hoping Ampache vote to increase chances of success.
